File: app/src/main/kotlin/metamark/parser/MatchZeroOrMoreExpressionTerm.kt
package metamark.parser

import metamark.parser.antlr.MetamarkParser

data class MatchZeroOrMoreExpressionTerm(
    val expression: ExpressionTerm,
): ExpressionTerm() {
    companion object {
        fun build(
            ctx: MetamarkParser.Expression_matchZeroOrMoreExpressionContext,
        ): MatchZeroOrMoreExpressionTerm {
            return MatchZeroOrMoreExpressionTerm(
                expression = ExpressionTerm.build(ctx.repeatedExpression)
            )
        }
    }
}

File: src/main/MatchZeroOrMoreExpressionTerm.ts
import {ExpressionTerm} from './ExpressionTerm';
import {Expression_matchZeroOrMoreExpressionContext} from "../generated_parser/MetamarkParser";

export class MatchZeroOrMoreExpressionTerm extends ExpressionTerm {
    static build(ctx: Expression_matchZeroOrMoreExpressionContext): MatchZeroOrMoreExpressionTerm {
        return new MatchZeroOrMoreExpressionTerm(ExpressionTerm.build(ctx._repeatedExpression));
    }

    readonly expression: ExpressionTerm;

    constructor(expression: ExpressionTerm) {
        super();
        this.expression = expression;
    }
}
